openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java

```diff
@@ -1165,6 +1173,13 @@ public Response deployApplicationFlow(
         IngestionPipeline ingestionPipeline = getIngestionPipeline(uriInfo, securityContext, app);
         ServiceEntityInterface service =
             Entity.getEntity(ingestionPipeline.getService(), "", Include.NON_DELETED);
+
+        if (app.getSupportsIngestionRunner()) {
+          if (service instanceof MetadataService) {
```

Comment:
why only the MetadataService?

Reply:
My assumption is apps have a MetadataService object. ServiceInterface does not have a ingestionRunner attribute, so it was mainly for safe casting. 

LMK if my assumption is incorrect